### PR TITLE
Add colours to rota

### PR DESCRIPTION
**Pull request by chriscarter90**
* Add colours to rota slots (based on the organisation's `uid`)
* Show empty dates when viewing the rota

### CODE REVIEW

File: app/models/rota.rb [hunk 1]
```diff
@@ -11,6 +11,18 @@ def to_partial_path
     "rotas/rota"
   end
 
+  def date_range
+    if !rota_slots.empty?
+      (sorted_slots.first.date..sorted_slots.last.date)
```
Comment (FriedSock) [inline app/models/rota.rb]: Do you need parentheses here?

File: app/views/rotas/_rota.html.erb [hunk 2]
```diff
@@ -14,19 +14,14 @@
...
       <td><%= date.strftime("%A, %-d %b %Y") %></td>
 
       <% @rota.shifts.each do |shift| %>
-        <td>
-        <%= slots.select { |s| s.shift_id == shift.id }.
-          map(&:organisation_uid).
-          map { |o| @rota.organisations.detect { |os| os.uid == o }}.
-          map(&:name).
-          join("</br></br>").
-          html_safe %>
```
Comment (FriedSock) [inline app/views/rotas/_rota.html.erb]: Shame to lose this monstrosity :person_frowning:

**Comment by pedrosmmoreira** [comment]
Not to crazy about the amount of logic in helpers here and the amount of params flying around between methods in the `Rota` model. I'm happy to merge this now but it feels like we should be able to come up with better "homes" and encapsulation of data. Any thoughts?

**Comment by chriscarter90** [comment]
@pedrosmmoreira Sure, am happy to look at this before merging. Any suggestions?

File: app/helpers/rota_helper.rb [hunk 3]
```diff
@@ -0,0 +1,51 @@
+module RotaHelper
+  def select_allocated_organisations(rota, shift, slots)
+    organisation_uids = slots.select { |s| s.shift_id == shift.id }.map(&:organisation_uid)
+
+    organisations = @rota.organisations_with_uids(organisation_uids)
```
Comment (pedrosmmoreira) [inline app/helpers/rota_helper.rb]: Should this be referencing the rota instance variable? If it is, why are we passing a rota parameter down to it?

Comment (chriscarter90) [inline app/helpers/rota_helper.rb]: I fixed that as part of the last commit - I noticed it too :)

File: app/views/rotas/_rota.html.erb (hunk 2 above)
Comment (chriscarter90) [inline app/views/rotas/_rota.html.erb]: :cactus:

**Comment by pedrosmmoreira** [comment]
LGTM

**Pull request merged**